viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
-# ------------- WAVE 1 --------------------
 
+# ------------- WAVE 1 --------------------
+# Creates a dictionary that holds movies with title, genre, and rating
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {"title": title,
+                 "genre": genre,
+                 "rating": rating}
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        return new_movie
+
+# Creates a dictionary for watched movies
+
+
+def add_to_watched(user_data, movie):
+    watched_move_list = {"watched": [movie]}
+    return watched_move_list
+
+# Creates a dictionary for movies in watchlist
+
+
+def add_to_watchlist(user_data, movie):
+    watchlist_list = {"watchlist": [movie]}
+    return watchlist_list
+
+# Removing watched movie from watchlist to watched within a dictionary
+
+
+def watch_movie(user_data, movie):
+    user_updated_data = user_data
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == movie:
+            remove_movie = user_updated_data["watchlist"].pop(i)
+            user_updated_data["watched"] += [remove_movie]
+    return user_updated_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# Calculates average rating for watched movies
+def get_watched_avg_rating(user_data):
+    counter = 0.0
+    for key, value in user_data.items():
+        for information in value:
+            counter += information["rating"]
+
+    length_of_movies = len(user_data["watched"])
+    print(length_of_movies)
+    if length_of_movies == 0:
+        return counter
+    else:
+        average = counter / length_of_movies
+        return average
+
+# Finds most watche genre in user's watched within dictionary
+
+
+def get_most_watched_genre(user_data):
+    most_watched_genre = {}
+    for movies in user_data["watched"]:
+        for values in movies:
+            if movies["genre"] not in most_watched_genre:
+                most_watched_genre[movies["genre"]] = 1
+            else:
+                most_watched_genre[movies["genre"]] += 1

```

Comment:
hmm what is this second for loop doing? It doesn't look like you are using `values` as a variable anywhere.  I think we have all the information we need with just the first for loop. The rest looks great!

```suggestion
        if movies["genre"] not in most_watched_genre:
            most_watched_genre[movies["genre"]] = 1
        else:
            most_watched_genre[movies["genre"]] += 1
```